lib/reek/smells/smell_warning.rb

```diff
@@ -88,25 +95,23 @@ def smell_classes
         [smell_detector.smell_category, smell_detector.smell_type]
       end
 
-      def common_parameters_equal?(other_parameters)
-        other_keys   = other_parameters.keys
-        other_values = other_parameters.values
+      def check_details_exist(other_details)
+        other_details.keys.each do |key|
+          unless DETAIL_KEYS.include?(key)
+            raise ArgumentError, "The detail #{key} you want to check for doesn't exist"
```

Comment:
Actually just see my comment directly above regarding "matches_smell_details?", that should make everthing clear ;)